backend/tests/test_excessive_db_polling.py

```
@@ -0,0 +1,146 @@
+    """Tests that the /health endpoint caches its DB query result."""
+
+    def test_health_check_returns_ok(self, client):
+        """Health check should return status ok."""
+        response = client.get("/health")
+        assert response.status_code == 200
+        data = response.json()
+        assert data["status"] == "ok"
+
+    def test_health_check_is_cached(self, client):
+        """Second call within TTL should return cached result without new DB query."""
+        # First call – populates cache
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)